### PR TITLE
fix YAML contentType handling in TransportShardUpsertAction

### PR DESCRIPTION
**Pull request by mfussenegger**
Since eabe917773b31cf624b187a6f4689f27e08484fa
`testCopyFromPartitionedTableCustomSchema` would fail when
`Requests.INDEX_CONTENT_TYPE` got set to YAML.

This fixes the issue and also avoids creating an unnecessary
XContentBuilder (which by default allocates 16kb) if the request
contains a rawSource

### CODE REVIEW

File: sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java [hunk 1]
```diff
@@ -321,22 +323,36 @@ public IndexRequest prepareUpdate(ShardUpsertRequest request, ShardUpsertRequest
...
-                .create(!request.overwriteDuplicates()).operationThreaded(false);
-        if (rawSource != null) {
-            indexRequest.source(rawSource.bytes);
+
+        BytesReference source;
+        if (rawSourceColumn > -1) {
+            source = new BytesArray((BytesRef) item.insertValues()[rawSourceColumn]);
         } else {
-            indexRequest.source(builder.bytes());
+            XContentBuilder builder = XContentFactory.jsonBuilder().startObject();
```
Comment (seut) [inline sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java]: just to be safe, pls use XContentFactory.contentBuilder(Requests.INDEX_CONTENT_TYPE)

**Comment by seut** [comment]
otherwise ack

File: sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/executor/transport/TransportShardUpsertAction.java]: will do in a separate PR and first try to figure out why we don't have any tests that fail because of this
